cypress/S3Monitor.py

```diff
@@ -0,0 +1,131 @@
+import boto3
+import json
+import os
+import quopri
+import time
+import re
+
+mailbox_bucket = 'opg-lpa-casper-mailbox' # this might get renamed once casper tests are turned off
+activation_emails_path = 'cypress/activation_emails'
+
+def assume_role_and_get_client():
+    sts = boto3.client(
+        'sts',
+        region_name='eu-west-1',
+    )
+
+    if os.getenv('CI'):
+        print("Assuming CI role")
+        role_arn = 'arn:aws:iam::050256574573:role/opg-lpa-ci'
+    else:
+        print("Assuming operator role")
+        role_arn = 'arn:aws:iam::050256574573:role/operator'
+
+    result = sts.assume_role(
+        RoleArn=role_arn,
+        RoleSessionName='session1',
+    )
+
+    s3Client = boto3.client(
+        's3',
+        aws_access_key_id=result['Credentials']['AccessKeyId'],
+        aws_secret_access_key=result['Credentials']['SecretAccessKey'],
+        aws_session_token=result['Credentials']['SessionToken']
+    )
+
+    return s3Client
+
+# Extract the plus part from emails of the form:
+# basename+pluspart@example.com
+def getPlusPartFromEmailAddress(email):
+    plusPos = email.find('+')
+    atPos = email.find('@')
+    userIdLength = atPos - plusPos - 1
+    userId = email[plusPos + 1:atPos]
+    return userId
+
```

Comment:
Suggest using a regex and groups instead of character positions, e.g.

```
import re

def getPlusPartFromEmailAddress(email):
    match = re.search("[^\+]\+(.+)@", email)
    if match is None:
        return ""
    return match.group(1)
```

Reply:
Thanks that works. :)  I had just replicated the slightly clumsy way done in the original PHP, but this is nicer!